recipe/build.sh

```diff
@@ -1,229 +1,3 @@
-        export USE_PRIORITIZED_TEXT_FOR_LD=1
-    fi
-    # Even though cudnn is used for CUDA builds, it's good to enable
-    # for MKLDNN for CUDA builds when CUDA builds are used on a machine
-    # with no NVIDIA GPUs.
-    export USE_MKLDNN=1
-    export USE_CUDA=1
-    export USE_CUFILE=1
-    # PyTorch has multiple different bits of logic finding CUDA, override
-    # all of them.
-    export CUDAToolkit_BIN_DIR=${BUILD_PREFIX}/bin
-    export CUDAToolkit_ROOT_DIR=${PREFIX}
-    if [[ "${target_platform}" != "${build_platform}" ]]; then
-        export CUDA_TOOLKIT_ROOT=${PREFIX}
-    fi
-    case ${target_platform} in
-        linux-64)
-            export CUDAToolkit_TARGET_DIR=${PREFIX}/targets/x86_64-linux
-            ;;
-        linux-aarch64)
-            export CUDAToolkit_TARGET_DIR=${PREFIX}/targets/sbsa-linux
-            ;;
-        *)
-            echo "unknown CUDA arch, edit build.sh"
-            exit 1
-    esac
-    case ${cuda_compiler_version} in
-        12.6)
-            export TORCH_CUDA_ARCH_LIST="5.0;6.0;6.1;7.0;7.5;8.0;8.6;8.9;9.0+PTX"
-            ;;
-        *)
-            echo "unsupported cuda version. edit build.sh"
-            exit 1
-    esac
-    export TORCH_NVCC_FLAGS="-Xfatbin -compress-all"
-    export NCCL_ROOT_DIR=$PREFIX
-    export NCCL_INCLUDE_DIR=$PREFIX/include
-    export USE_SYSTEM_NCCL=1
-    export USE_STATIC_NCCL=0
-    export USE_STATIC_CUDNN=0
-    export MAGMA_HOME="${PREFIX}"
-else
-    if [[ "$target_platform" != *-64 ]]; then
-      # Breakpad seems to not work on aarch64 or ppc64le
-      # https://github.com/pytorch/pytorch/issues/67083
-      export USE_BREAKPAD=0
-    fi
-    # MKLDNN is an Apache-2.0 licensed library for DNNs and is used
-    # for CPU builds. Not to be confused with MKL.
-    export USE_MKLDNN=1
-    export USE_CUDA=0
-fi
-
-echo '${CXX}'=${CXX}
-echo '${PREFIX}'=${PREFIX}
-$PREFIX/bin/python -m pip $PIP_ACTION . --no-deps -vvv --no-clean \
-    | sed "s,${CXX},\$\{CXX\},g" \
-    | sed "s,${PREFIX},\$\{PREFIX\},g"
-
-if [[ "$PKG_NAME" == "libtorch" ]]; then
-  mkdir -p $SRC_DIR/dist
-  pushd $SRC_DIR/dist
-  wheel unpack ../torch-*.whl
-  pushd torch-*
-  mv torch/bin/* ${PREFIX}/bin
-  mv torch/lib/* ${PREFIX}/lib
-  mv torch/share/* ${PREFIX}/share
-  for f in ATen caffe2 tensorpipe torch c10; do
-    mv torch/include/$f ${PREFIX}/include/$f
-  done
-  rm ${PREFIX}/lib/libtorch_python.*
-  popd
-  popd
-
-  # Keep the original backed up to sed later
-  cp build/CMakeCache.txt build/CMakeCache.txt.orig
-else
-  # Keep this in ${PREFIX}/lib so that the library can be found by
-  # TorchConfig.cmake.
-  # With upstream non-split build, `libtorch_python.so`
-  # and TorchConfig.cmake are both in ${SP_DIR}/torch/lib and therefore
-  # this is not needed.
-  mv ${SP_DIR}/torch/lib/libtorch_python${SHLIB_EXT} ${PREFIX}/lib
-fi
+# we are using a separate file here to avoid conda-build thinking that
```

Comment:
Is a separate build_common.sh still necessary if we don’t end up having non-magma builds?